### PR TITLE
remove unused struct field `format_version` on Snapshot

### PR DESCRIPTION
**Pull request by Junnplus**
# Reason for This PR

`[Author TODO: add issue #.]`
`[Open the PR after the related issue has a clear conclusion.]`
`[If there is no issue which states the need for this PR, create one first.]`
Fixes https://github.com/firecracker-microvm/firecracker/issues/2953

## Description of Changes

`[Author TODO: add description of changes.]`

- [ ] This functionality can be added in [`rust-vmm`][1].

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The issue which led to this PR has a clear conclusion.
- [x] This PR follows the solution outlined in the related issue.
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] Any newly added `unsafe` code is properly documented.
- [x] Any API changes follow the [Runbook for Firecracker API changes][2].
- [x] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.

[1]: https://github.com/rust-vmm
[2]: ../docs/api-change-runbook.md

### CODE REVIEW

**Review by dianpopa — COMMENTED** [review]
Indeed, the `format_version` field is not used, since the version of the snapshot's format is encoded in the [magic value found in the snapshot header](https://github.com/firecracker-microvm/firecracker/blob/main/src/snapshot/src/lib.rs#L114).

In order for the tests to pass you need to:
* also get rid of the [const](https://github.com/firecracker-microvm/firecracker/blob/main/src/snapshot/src/lib.rs#L35) 
* limit commit title to 50 chars, checkout the 50/72 formatting guideline

**Comment by alindima** [comment]
Need to update the target coverage as well in order to have the test pass, [here](https://github.com/firecracker-microvm/firecracker/blob/a0e22daa74571d0aeabae766139aa9ab3ea7bebb/tests/integration_tests/build/test_coverage.py#L32) to `84.48`

**Comment by alindima** [comment]
Unfortunately, since there's been another merge on the main branch, you need to update coverage again 😄

File: tests/integration_tests/build/test_coverage.py [hunk 1]
```diff
@@ -29,7 +29,7 @@
 # Checkout the cpuid crate. In the future other
 # differences may appear.
 if utils.is_io_uring_supported():
-    COVERAGE_DICT = {"Intel": 85.18, "AMD": 84.66, "ARM": 84.24}
+    COVERAGE_DICT = {"Intel": 85.18, "AMD": 84.66, "ARM": 84.48}
```
Comment (AlexandruCihodaru) [inline tests/integration_tests/build/test_coverage.py:32]: Here the coverage should be set to 84.22 in order for build to pass

**Review by AlexandruCihodaru — APPROVED** [review]

**Review by alindima — APPROVED** [review]

**Pull request merged**